atc/db/build.go

```diff
@@ -1584,9 +1585,10 @@ func (b *build) AdoptRerunInputsAndPipes() ([]BuildInput, bool, error) {
 		err = psql.Select("v.version").
 			From("resource_config_versions v").
 			Join("resources r ON r.resource_config_scope_id = v.resource_config_scope_id").
-			Where(sq.Eq{
-				"v.version_sha256": input.Input.Version,
-				"r.id":             input.Input.ResourceID,
+			Where(sq.Eq{"r.id": input.Input.ResourceID}).
+			Where(sq.Or{
+				sq.Eq{"v.version_md5": input.Input.Version},
+				sq.Eq{"v.version_sha256": input.Input.Version},
```

Comment:
nit: could optimize by checking the length of `input.Input.Version` and figuring out which column I need to query.